### PR TITLE
Fix for #154, added documentation

### LINKED ISSUE #154: Create doc strings for UserDao class and its functions
## Description
As a developer,
I need to have in code documentation,
so that I can understand better the code.

Related to #149 

We need to have **doc strings for UserDao class and its functions**. You can find this file [app/api/dao/user.py](https://github.com/systers/mentorship-backend/blob/develop/app/api/dao/user.py).
You can follow these conventions: [style guide for doc strings in python by Google](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings).

### Update [Required]
- [ ] Create doc strings for UserDao class and its functions (file: app/api/dao/user.py).

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hours

### PR DESCRIPTION
**Pull request by amnoweb**
### Description
Added doc strings to the code explaining the UserDAO class and its functions, following python doc string conventions.

Fixes #154 

### Type of Change:
- Documentation

### How Has This Been Tested?
Did not require tests, as it was not a change to the functioning code but simply documentation.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand 
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/api/dao/user.py [hunk 1]
```diff
@@ -9,12 +9,25 @@
...
     def create_user(data):
+        """Creates a new user.
+        
+        Creates a new user with provided data.
+        
+        Arguments:
+            data: A list containing the user's name, username, password, and email, as well as recognition that they have read and agree to the Terms and Conditions.
+            
+        Returns:
+            A list with two elements. The first element is a dictionary containing a key 'message' containing a string which indicates whether or not the user was created successfully. The second is the HTTP response code.
```
Comment (isabelcosta) [inline app/api/dao/user.py]:
```suggestion
            A tuple with two elements. The first element is a dictionary containing a key 'message' containing a string which indicates whether or not the user was created successfully. The second is the HTTP response code.
```

What is returned is a tuple and not a list.

File: app/api/dao/user.py [hunk 2]
```diff
@@ -43,6 +56,17 @@ def create_user(data):
...
     def delete_user(user_id):
+        """ Deletes a user.
+        
+        Deletes the specified user and removes them from the directory, with checks to make sure that the user exists and is not the only administrator.
+        
+        Arguments:
+            user_id: The ID of the user to be deleted.
+            
+        Returns: 
+            A list with two elements. The first element is a dictionary containing a key 'message' containing a string which indicates whether or not the user was created successfully. The second is the HTTP response code.
```
Comment (isabelcosta) [inline app/api/dao/user.py]:
```suggestion
            A tuple with two elements. The first element is a dictionary containing a key 'message' containing a string which indicates whether or not the user was created successfully. The second is the HTTP response code.
```

What is returned is a tuple and not a list.

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@amnoweb thank you for this contribution!
I left two change suggestions. Let me know when you change it and feel free to discuss if you have any better suggestion.

**Comment by amnoweb** [comment]
@isabelcosta, your changes looked good so I just committed them, hopefully that works!

**Review by isabelcosta — APPROVED** [review]
Thank you @amnoweb for doing these changes 🎉

**Pull request merged**